eqt/ui/FormDialog.py

```diff
@@ -309,3 +322,115 @@ def applyWidgetStates(self, states):
+        parent based on the values of the widgets in the advanced dialog.
+        """
+        self.saveAllWidgetStates()
+        self.onOk()
+        self.formWidget.setDefaultWidgetStatesVisibleTrue()
+        self.close()
+
+        if self.display_on_parent_dict:
+            if self.getSavedWidgetStates() == self.getDefaultWidgetStates():
+                self._removeWidgetsFromParent()
+            else:
+                self._addOrUpdateWidgetsInParent()
+
+    def _addOrUpdateWidgetsInParent(self):
+        """
+        Adds or updates widgets in the parent form.
+
+        This method iterates over the display_on_parent_dict and adds the widgets to the parent
+        form. If a widget already exists in the parent form, it is updated with the most current
+        value set in the advanced dialog.
+        """
+        for index, name in enumerate(self.display_on_parent_dict, start=1):
+            widget_row = self.parent_button_row + index if self.parent_button_row != -1 else -1
+            value = str(self.getSavedWidgetStates()[f'{name}_field']['value'])
+            if f'{name}_field' not in self.dialog_parent.getWidgets():
+                label = str(self.getSavedWidgetStates()[f'{name}_label']['value'])
+                self.dialog_parent.insertWidget(widget_row, name, QtWidgets.QLabel(value), label)
+            else:
+                self.dialog_parent.getWidget(name, 'field').setText(value)
+
+    def _removeWidgetsFromParent(self):
+        """
+        Removes widgets from the parent form.
+
+        This method iterates over the display_on_parent_dict and removes
+        the widgets from the parent.
+        """
+        for name in self.display_on_parent_dict:
+            if f'{name}_field' in self.dialog_parent.getWidgets():
+                self.dialog_parent.removeWidget(name)
+
+    def setDefaultValues(self):
+        """
+        Sets the widgets in the advanced dialog to their default states.
+
+        Makes the default widget states visible and applies the widget states
+        to the widgets in the form.
+        """
+        self.formWidget.setDefaultWidgetStatesVisibleTrue()
+        self.applyWidgetStates(self.formWidget.default_widget_states)
+
+    def addToDictionaryDisplayOnParent(self, name):
+        """
+        Adds `name` in a dictionary. The order in which names are added to this
+        dictionary reflects the order in which the widgets are added to the parent.
+        Raises an error if the parent of the advanced dialog is None.
+
+        Parameters
+        ----------
+        name : str
+            The name of the widget in the advanced dialog to be displayed in the parent.
+        """
+        if self.dialog_parent is None:
+            raise KeyError('''The advanced-dialog parent is None.
+                           Set the parent if you want to add widgets to it.''')
+        self.display_on_parent_dict[name] = None
```

Comment:
This could be a `list` rather than a `dictionary`.

Reply:
done